app/models/ability.rb

```diff
@@ -0,0 +1,36 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the user here. For example:
+    #
+    #   return unless user.present?
+    #   can :read, :all
+    #   return unless user.admin?
+    #   can :manage, :all
+    #
+    if user.is?(:admin)
+      can :manage, :all
+    else
+      can :destroy, Post, author: user
+      can :destroy, Comment, author: user
+    end
+    # The first argument to `can` is the action you are giving the user
+    # permission to do.
+    # If you pass :manage it will apply to every action. Other common actions
+    # here are :read, :create, :update and :destroy.
+    #
+    # The second argument is the resource the user can perform the action on.
+    # If you pass :all it will apply to every resource. Otherwise pass a Ruby
+    # class of the resource.
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, published: true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md
+  end
+end
```

Comment:
- [ ] Its great that you have made the is? method in your user model 🥇 but it would be nice if you see all the cases in your app. You didn't add any authorization for a user who is not signed in means a guest user 👍 Kindly do that 😊
- [ ] A user who is not a admin can be read all blogs. 🤔 You should cover this scenario as well 😊

Hint: You can see the below code to get an idea what i am talking about 👍

```ruby
 class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end

```

Reply:
@HAMMAS-SALEEM thank you for your feedback. That's a good note for a `new user`, although I don't remember being mentioned in the project requirements, I will add it since it's a good improvement. 
Does it mean I will need to add a view for new user?

Reply:
Hey @michael-duke I hope you are doing well. Thanks for pointing out that it is not mentioned in your requirements.
But why i suggested it because it is gonna give you an error when there will be no users. It has nothing to do with creating a new view. You can check it by doing it. You can try your application without add this whole line when there will be no users.

`user ||= User.new`

You will get an error that user don't exist. To prevent that error you need to add the above line to tell your application that if there is no user then just create a guest user.

You can relate it to youtube. On youtube you don't need to sign up to watch videos because youtube makes you a guest user but for commenting and liking a video you need to create an account. So think about that example and you can relate that example to this scenario :blush:

Reply:
It's clear now @HAMMAS-SALEEM thanks for the update on that part so this scenario covers when I have no users in my database, I see now

Reply:
Thank you @HAMMAS-SALEEM  for meeting me on a call. I learned so much about `devise` methods and `cancan`. I think my app will be complete with the changes we discussed the `guest user` and his permissions. It has been a great session and constructive feedback